include/drivers/modem/gsm_ppp.h

```diff
@@ -60,4 +61,17 @@ void gsm_ppp_register_modem_power_callback(const struct device *dev,
  */
 const struct gsm_ppp_modem_info *gsm_ppp_modem_info(const struct device *dev);
 
+/**
+ * @brief Set custom modem setup commands.
+ *
+ * @param dev: gsm modem device.
+ * @param data: custom modem setup commands array.
+ * @param len: length of custom modem setup commands array.
+ *
+ * @retval None.
+ */
+void gsm_ppp_set_custom_setup_cmds(const struct device *dev,
+				   const struct setup_cmd *data,
+				   size_t len);
```

Comment:
I do not have big preference here, using callbacks sounds a bit more flexible so I am inclined to that direction. If both options are possible, then +1 to that.

Reply:
@mniestroj what is your opinion about having both options? Currently I would be able to get some time to finish at least the current proposition so it's still better to have this one instead of nothing.